From 59e1bf06e8bc2379db5389fee3c45c35f20dc4ac Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Thu, 10 May 2012 17:29:51 +0300
Subject: [PATCH 163/311] conf: add dependency on extended db/db.h

Colibri DB interfaces (db/db.h) should be extended to allow user-space
applications to work with in-memory Berkeley DB.  This is needed by confd
and user-space confc.
---
 conf/map.h | 15 +++------------
 conf/obj.c |  7 +++++++
 2 files changed, 10 insertions(+), 12 deletions(-)

diff --git a/conf/map.h b/conf/map.h
index d768d23..5a0a27c 100644
--- a/conf/map.h
+++ b/conf/map.h
@@ -27,18 +27,9 @@
  *
  * XXX @todo Explain what is this registry (AKA objid-to-address mapping) for.
  *
- * The implementation of c2_conf_map is based on in-memory Berkeley DB.
- * Since it is not possible to create a "memory-only" Berkeley
- * database using current version of Colibri's database interfaces
- * ("db/db.h"), this implementation invokes Berkeley DB API (<db.h>)
- * directly.
- *
- * DB environments (DB_ENV *) are not used: a registry of cached
- * objects is never accessed by several threads, so there is no use
- * for multi-threaded features provided by a DB environment.
- *
- * @see Writing In-Memory Berkeley DB Applications
- * [http://docs.oracle.com/cd/E17076_02/html/articles/inmemory/C/index.html]
+ * The implementation of c2_conf_map is based on in-memory Berkeley
+ * DB, which is created and manipulated via Colibri database
+ * interfaces ("db/db.h").
  *
  * @see conf_dfspec_map "Detailed Functional Specification"
  */
diff --git a/conf/obj.c b/conf/obj.c
index 03e3a7c..9ce8d02 100644
--- a/conf/obj.c
+++ b/conf/obj.c
@@ -145,6 +145,13 @@
  *   assumption and load the entire contents of configuration database
  *   into the cache.
  *
+ * - Colibri database library ("db/db.h") should provide a user-space
+ *   interface for creating in-memory databases. This interface will
+ *   be used by confd and user-space confc; see @ref conf-fspec-map.
+ *
+ *   See also `Writing In-Memory Berkeley DB Applications'
+ *   [http://docs.oracle.com/cd/E17076_02/html/articles/inmemory/C/index.html].
+ *
  * - c2_rpc_item_get() and c2_rpc_item_put() should be implemented.
  *
  *   Confc implementation schedules a state transition in
-- 
1.8.3.2

